zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -103,6 +108,17 @@
   // ZkEventThread. Otherwise the retry request might block the normal event processing.
   protected final ZkAsyncRetryThread _asyncCallRetryThread;
 
+  static {
```

Comment:
Please add a TODO item here to state this is just a workaround fix because zookeeper does not return the failure reason if it fails to read a large children list.

Reply:
I added a TODO at `getChildren()` so when we remove that code part at getChildren(), we will also remove this private LIMIT. To make it clear, I also add a TODO for `NUM_CHILDREN_LIMIT`.